content/local-setup.md

```diff
@@ -0,0 +1,359 @@
+
+### Create a new project
+
+To create a new Rust package to write a Solana program, you can use the `cargo new --lib` command with the name of the new directory you'd like to create.
+
+```bash
+cargo new --lib <PROJECT_DIRECTORY_NAME>
+```
+
+This command will create a new directory with the name you specified at the end of the command. This new directory will contain a `Cargo.toml` manifest file that describes the package.
+
+The manifest file contains metadata such as name, version, and dependencies (crates). To write a Solana program, you’ll need to update the `Cargo.toml` file to include `solana-program` as a dependency. You may also need to add the `[lib]` and `crate-type` lines shown below.
+
+```rust
+[package]
+name = "<PROJECT_DIRECTORY_NAME>"
+version = "0.1.0"
+edition = "2021"
+
+[features]
+no-entrypoint = []
+
+[dependencies]
+solana-program = "~1.8.14"
+
+[lib]
+crate-type = ["cdylib", "lib"]
+```
+
+At that point, you can start writing your program in the `src` folder.
+
+### Build and deploy
+
+When it comes time to build your Solana program, you can use the `cargo build-bpf` command.
+
+```bash
+cargo build-bpf
+```
+
+The output of this command will include instructions for a deploying your program that look something like this:
+
+> To deploy this program:
+>   $ solana program deploy /Users/James/Dev/Work/solana-hello-world-local/target/deploy/solana_hello_world_local.so
+> The program address will default to this keypair (override with --program-id):
+>   /Users/James/Dev/Work/solana-hello-world-local/target/deploy/solana_hello_world_local-keypair.json
+
+When you are ready to deploy the program, use the `solana program deploy` command output from `cargo build-bpf`. This will deploy your program to the cluster specified in your CLI configuration.
+
+```rust
+solana program deploy <PATH>
+```
+
+# Demo
+
+Let's practice by building and deploying the "Hello World!" program that we created in the [Hello World lesson](https://github.com/Unboxed-Software/solana-course/pull/content/hello-world-program.md).
+
+We'll do this all locally, including deploying to a local test validator. Before we begin, make sure you've installed Rust and the Solana CLI. You can refer to the instructions in the overview to get set up if you haven't already.
+
+### 1. Create a new Rust project
+
+Let's start by creating a new Rust project. Run the `cargo new --lib` command below. Feel free to replace the directory name with your own.
+
+```bash
+cargo new --lib solana-hello-world-local
+```
+
+Remember to update the `cargo.toml` file to include `solana-program` as a dependency and the `crate-type` if isn't there already.
+
+```bash
+[package]
+name = "solana-hello-world-local"
+version = "0.1.0"
+edition = "2021"
+
+[dependencies]
+solana-program = "~1.8.14"
```

Comment:
should the version be updated or does it not matter for demonstration purposes?

Reply:
I couldn't get hello world to build with the newer version, so I just used this...mostly out of laziness and not wanting to figure out the problem haha

Reply:
maybe a mismatch of my rust version and the solana-program version?

Reply:
It also looks like the latest solana-program version is higher than what you had put before also @ZYJLiu 

On the rust site for the crate it says v 1.11.3

Reply:
This is the error I get every time I try to use a more recent version of solana-progra
![Screen Shot 2022-07-18 at 12 42 13 PM](https://user-images.githubusercontent.com/11344110/179580544-b9e8526c-817a-4463-bd11-4e5a99709d56.png)
m